### ISSUE TITLE
Worker behavior Moneroocean 

### ISSUE THREAD

**Issue by anbu-squad** [UnamSanctam/SilentCryptoMiner#2055]
I've been observing strange behavior of workers on the Monerocean pool for several days now.
Workers seem to switch to another pool from time to time

I read questions from other users here, but I didn’t seem to find anything strange on my panel

Question in general, regarding the remote configuration
Perhaps there are methods to influence workers using remote configuration? Otherwise, I simply cannot explain to myself the strange behavior of workers

I also made a modest donation
In my strange behavior of workers, I observed a relatively good hashrate on the moenrookean pool

**Comment by UnamSanctam** [comment]
Thank you. Do you use the hashrate splitting feature on MoneroOcean or just normal mining? It's also possible that your miners are unlucky sometimes (in finding shares) which can cause gaps in the statistics (since the pool only shows a miner when it submits a share). And it's also possible that some of your miners are getting temporarily banned by MoneroOcean for 15 minutes, it can happen sometimes on MoneroOcean, at least in my experience when testing things. You can also try setting "keepalive" to true if it isn't, that can prevent timeouts from the pool, though it's also possible that "keepalive" can cause problems depending on the pool.

**Comment by anbu-squad** [comment]
```
[2023-12-23 19:09:00.331] [gulf.moneroocean.stream:443] timeout
[2023-12-23 19:09:23.340] [gulf.moneroocean.stream:443] timeout
[2023-12-23 19:09:23.341] [gulf.moneroocean.stream:443] timeout
```

```
[2023-12-23 19:12:25.136] [gulf.moneroocean.stream:20128] timeout
[2023-12-23 19:12:48.141] [gulf.moneroocean.stream:20128] timeout
[2023-12-23 19:12:50.142] [gulf.moneroocean.stream:20128] timeout

```


```

{
	"algo": "rx/0",
	"pool": "gulf.moneroocean.stream",
        "port": 443,
        "wallet": "wallet.{COMPUTERNAME}",
        "password": "",
	"nicehash": false,
	"ssltls": true,
	"max-cpu": 30,
	"idle-wait": 5,
	"idle-cpu": 80,
	"stealth": true,
    "stealth-fullscreen": true,
	"stealth-targets": "Taskmgr.exe,ProcessHacker.exe,perfmon.exe,procexp.exe,procexp64.exe,WorldOfTanks.exe,wotblitz.exe",
    "kill-targets": "afwServ.exe"
}

```


what else can be added to the remote configuration block? hash rate sharing functions, I did not use
I don't even know how to use the function

can someone else influence the work of miners?
through the panel, sending your configurations to miners for example

**Comment by UnamSanctam** [comment]
You can try adding `"keepalive": true` and see if it helps in an way. Though timeouts on pools (in MoneroOcean as well) do happen sometimes. If someone had some temporary control over your miners (can really only happen if someone hacks your web panel or "Remote Configuration") then they would usually change your configuration to theirs and not just switch some miners configurations shortly from time to time.

**Comment by anbu-squad** [comment]
In general, I try to adhere to safety rules

Also, how does the hashrate distribution function turn on?

in the remote configuration block, can we simply change the command line arguments of the miner itself?
I just want to understand the essence for myself

**Comment by UnamSanctam** [comment]
The hashrate splitting option would be inside the wallet address, so if your wallet address is in the format that you sent previously then it's not on. How often do the disconnects happen and for how long? And how do you know that they have disconnected, and where do you see it, in the web panel?

>in the remote configuration block, can we simply change the command line arguments of the miner itself?
I just want to understand the essence for myself

No I have to manually implement any options that should be changeable with the "Remote Configuration" or web panel due to how computers and programs work.

**Comment by anbu-squad** [comment]
I checked the web panel earlier, I didn’t seem to notice anything suspicious in terms of other wallets or pools

in general, for several days now my conventional 250 kh/s has dropped to 7 kh/s
also with the number of workers, from 120 to 5 in a short period of time

2 to 10 minutes pass and the hashrate and workers return to mining on the pool again

but this is me figuratively
I probably should have recorded a video =)

and one more nuance
I would like to see new workers on the pool when they first connect
I put {computer name}-root-
after receiving data from the remote configuration, the worker name is already without -root-

So the whole point is that the workers I know by name again receive the prefix -root-
which is specified in the build settings
after being away from the pool

**Comment by anbu-squad** [comment]
I've already started to think
maybe it's the pool itself that's at fault

and they are doing something there

**Comment by anbu-squad** [comment]
I had from 350 to 550 kh/s
then these dances began

and there was a cascading decline in the hashrate

**Comment by UnamSanctam** [comment]
All miners disconnect at about the same time? Then it sounds like the pool might be temporarily banning you or something might be happening with their pool server(s). Honestly the most consent pool  of the ones I've been trying so far is nanopool, though the statistics are quite slow there.

>and one more nuance
I would like to see new workers on the pool when they first connect
I put {computer name}-root-
after receiving data from the remote configuration, the worker name is already without -root-
So the whole point is that the workers I know by name again receive the prefix -root-
which is specified in the build settings
after being away from the pool

Does the pool support the `-` character? And do you have a configuration in the web panel? Since the miner will use that if you do.

**Comment by anbu-squad** [comment]
https://imgur.com/a/Lh5yPku

**Comment by anbu-squad** [comment]
yes, I installed the remote configuration in the web panel as well

-root- this is only in the settings of the build itself
purely because there is a new miner in the family

**Comment by UnamSanctam** [comment]
Aha alright, since so many got that weird pattern at the same time then it's most likely some pool issue (since the common factor of the miners is the pool).

**Comment by anbu-squad** [comment]
Thank you
I have temporarily moved to another pool

I'll leave the topic open for now
maybe someone will have a similar problem
